lib/src/query_component.dart

```diff
@@ -22,47 +22,69 @@ import 'util.dart';
   directives: [bsAccordionDirectives, BsInput, coreDirectives, formDirectives],
 )
 class QueryComponent implements OnInit {
-  // Names used in css.
+  // Values used in html.
+  static const queryTypeChoices = QueryType.values;
   static const requestSectionTitle = 'Request Parameter';
   static const advertiserParameterName = 'advertiserId';
   static const mediaPlanParameterName = 'campaignId';
-  static const numberOnlyPattern = '^[0-9]*\$';
   static const insertionOrderParameterName = 'insertionOrderId';
+  static const numberOnlyPattern = '^[0-9]+\$';
   static const underpacingCheckBoxName =
       'Highlight underpacing insertion orders';
   static const populateButtonName = 'populate';
-  static const parameterAlertText = 'Missing required parameter values.';
-  static bool showAlert = false;
-  static bool showSpinner = false;
 
-  final QueryService _queryService;
-  final ExcelDart _excel;
+  // Stores accordion panel selection, default is byAdvertiser.
+  QueryType queryType = QueryType.byAdvertiser;
 
-  // Default selection is byAdvertiser.
-  static QueryType queryType = QueryType.byAdvertiser;
-  static List<QueryType> queryTypeChoices = QueryType.values;
+  // Stores input box values.
+  String advertiserId;
+  String mediaPlanId;
+  String insertionOrderId;
 
-  static String advertiserId;
-  static String mediaPlanId;
-  static String insertionOrderId;
+  // Stores checkbox value, default is false.
+  bool highlightUnderpacing = false;
 
-  static bool highlightUnderpacing = false;
+  // Controls when to show the spinner on the populate button.
+  bool showSpinner = false;
+
+  final QueryService _queryService;
+  final ExcelDart _excel;
 
   QueryComponent(this._queryService, this._excel);
 
   @override
   void ngOnInit() async => await _excel.loadOffice();
 
-  void onClick() async {
-    // Determines if all required parameters are filled in.
-    if (_missingRequiredParameters()) {
-      showAlert = true;
-      showSpinner = false;
-      return;
+  /// Disables the populate button if missing required input ids or
+  /// if the ids are not integers.
+  ///
+  /// Binds to the [disabled] attribute on the populate button.
+  bool disablePopulateButton() {
+    var invalid = false;
+    final numberOnly = RegExp(numberOnlyPattern);
+    switch (queryType) {
+      case QueryType.byMediaPlan:
+        invalid = mediaPlanId == null || !numberOnly.hasMatch(mediaPlanId);
+        continue checkAdvertiser;
```

Comment:
woah thats cool

does io not need media plan id?

Reply:
haha nope :)